components/datetime/src/pattern/mod.rs

```diff
@@ -281,3 +288,113 @@ impl Serialize for Pattern {
         }
     }
 }
+
+/// Used to represent either H11/H12, or H23/H24. Skeletons only store these
+/// hour cycles as H12 or H24.
+#[derive(Debug, PartialEq, Clone, Copy)]
+#[cfg_attr(
+    feature = "provider_serde",
+    derive(serde::Serialize, serde::Deserialize)
+)]
+pub enum FlexibleHourCycle {
```

Comment:
Nit: Hmm, I'm not sure I like the name `FlexibleHourCycle`.  It sounds a bit too close to "flexible day period", which is a whole other beast.  Bikeshed:

- `PatternHourCycle` or `HourCycleForPattern`
- `CoarseHourCycle`
- `FormatHourCycle`

Note that ECMA-402 uses the language "pattern" (h23/h24) and "pattern12" (h11/h12) to distinguish.

https://tc39.es/ecma402/#sec-intl.datetimeformat-internal-slots

Reply:
I like `CoarseHourCycle`.